src/jsonrpccpp/server/connectors/httpserver.cpp

```diff
@@ -41,13 +42,62 @@ IClientConnectionHandler *HttpServer::GetHandler(const std::string &url) {
   return NULL;
 }
 
+namespace
+{
+
+class AddressInfo
```

Comment:
I think the struct is unnecessary. I'd suggest a simple method ` bool SetAddrInfo(const std::string& addr, const int port)`. Which sets an `struct addrinfo*` instance variable of HttpServer.

Reply:
The class was necessary to properly manage the `struct addrinfo` with RAII in the previous version.  I've now moved the variable to be managed by the `HttpServer` itself as you suggested, so now the utility class is indeed no longer needed.

Note that I've kept the method as `SetBindAddress` - that seems to be the simplest change to me, as the port is already provided in the constructor (mandatorily).  Thus providing it again to this method would be redundant and cause confusion as to which port is used.  Let me know if you want me to change that to exactly your suggested method.